src/common/snippets/include/snippets/runtime_configurator.hpp

```diff
@@ -157,58 +157,44 @@ class RuntimeConfigurator {
      */
     std::vector<std::vector<size_t>> extract_layouts() const;
 
-    class ParallelWAOptimizer {
+    class MHAParallelWAOptimizer {
     public:
+        MHAParallelWAOptimizer(RuntimeConfigurator* configurator);
         /**
-         * @brief Inits ParallelWAOptimizer: computes optimizer parameters which should be set at compilation stage
+         * @brief Inits MHAParallelWAOptimizer: computes optimizer parameters which should be set at compilation stage
          * @param linear_ir LinearIR
-         * @param io_descs Input/output descriptors which are used for optimizer parameters initialization
-         * @param in_num Number of inputs. It is needed to distinguish input and output shapes/layouts
          */
-        void init(const ov::snippets::lowered::LinearIRCPtr& linear_ir,
-                  const std::vector<snippets::lowered::PortDescriptorPtr>& io_descs,
-                  size_t in_num);
+        void init(const ov::snippets::lowered::LinearIRCPtr& linear_ir);
         /**
          * @brief Checks if optimizer is enabled
          * @todo Ticket 148891: when RuntimeConfigurator::update will be rewritten on PassPipeline, this method should be removed
-         * We will not just register ParallelWAOptimizer in case if it is not needed
+         * We will not just register MHAParallelWAOptimizer in case if it is not needed
          */
-        bool enabled();
+        bool enabled() const;
```

Comment:
It can be moved to private section now since it's called only in `optimize()`.

Thus, if we have ctor (instead of `init`) and just the one method `optimize` - it will be simple interface of the optimizer 😃 

```
class MHAParallelWAOptimizer {
   public:
       MHAParallelWAOptimizer(const LinearIRPtr& lir, const RuntimeConfiguratorPtr& configurator);

       bool optimize();
   private:
        .....
};
```